src/libs/actions/Report.js

```diff
@@ -1580,6 +1580,61 @@ Onyx.connect({
     },
 });
 
+/**
+ * Creates an optimistic IOU reportAction
+ *
+ * @param {String} type IOUReportAction type. Can be oneOf(create, decline, cancel, pay).
+ * @param {Number} amount IOU amount in cents.
+ * @param {String} comment user comment for the IOU.
+ * @param {String} paymentType passed only for IOUReportActions with type = pay IOU. Can be oneOf(elsewhere, payPal).
+ * @param {String} existingIOUTransactionID passed only for IOUReportActions with type = oneOf(cancel, decline). Generates a randomID as default.
+ * @param {Number} existingIOUReportID passed only for IOUReportActions with type = oneOf(decline, cancel, pay). Generates a randomID as default.
+ *
+ * @returns {Object}
+ */
+function createIOUReportAction(type, amount, comment, paymentType = '', existingIOUTransactionID = '', existingIOUReportID = 0) {
```

Comment:
I'm not sure if we should be adding this in Report.js, ReportAction.js or IOU.js. Let me know if any of you have a strong opinion and we can move this to a different file

Reply:
Agree with `buildOptimisticReportAction`. I prefer build but couldn't think of a good reason why until seeing this comment 👍

Reply:
I'd be happy to see a new file for building the optimistic data, but I'm not sure that would be a popular opinion. As long as we keep the three manualRequests build data functions together I don't think it matters too much.

Reply:
Renamed and moved to ReportUtils